records/state/state.go

```diff
@@ -200,6 +208,20 @@ func statusIPs(st []Status, src func(*Status) []string) []string {
 	return nil
 }
 
+// MapPort returns the mapped port if available, or the host port
+// listening on.
+func MapPort(t Task, hostport int) int {
+	ni := t.Statuses[0].ContainerStatus.NetworkInfos
+	for n := range ni {
+		for m := range ni[n].PortMappings {
+			if ni[n].PortMappings[m].HostPort == hostport {
+				return ni[n].PortMappings[m].ContainerPort
```

Comment:
possible edge case: if container port == 0 do we really want to use it? or should we fall back to host port? it's probably a framework bug to indicate a container port of zero. maybe we should log such cases and return the host port instead?

Reply:
For container port == 0, using the hostport seems fine.